drivers/mrf24j40/mrf24j40_getset.c

```diff
@@ -0,0 +1,511 @@
+                            54, 54, 54, 54, 54, 54, 53, 53, \
+                            53, 53, 53, 53, 53, 52, 52, 52, \
+                            52, 52, 52, 51, 51, 51, 51, 51, \
+                            50, 50, 50, 50, 50, 49, 49, 49, \
+                            49, 49, 48, 48, 48, 48, 47, 47, \
+                            47, 47, 47, 46, 46, 46, 46, 45, \
+                            45, 45, 45, 44, 44, 44, 44, 44, \
+                            43, 43, 43, 42, 42, 42, 42, 41, \
+                            41, 41, 41, 41, 41, 40, 40, 40, \
+                            40, 40, 39, 39, 39, 39, 39, 38, \
+                            38, 38, 38, 37, 37, 37, 36, 30 };
+
+
+    tmp = mrf24j40_reg_read_short(dev, MRF24J40_REG_CCAEDTH);       /* Energy detection threshold */
+
+    return(dBm_value[tmp]);                                     /* in dBm */
+}
+
+void mrf24j40_set_cca_threshold(mrf24j40_t *dev, int8_t value)
+{
+    /* take a look onto datasheet table 3-8 */
+    uint8_t RSSI_value[] = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, \
+                             0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, \
+                             0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, \
+                             0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfe, 0xfd, 0xfa, 0xf5, \
+                             0xef, 0xe9, 0xe4, 0xe1, 0xdd, 0xd8, 0xd4, 0xcf, 0xcb, 0xc6, \
+                             0xc1, 0xbc, 0xb7, 0xb0, 0xaa, 0xa5, 0x9f, 0x99, 0x94, 0x8f, \
+                             0x8a, 0x85, 0x81, 0x7d, 0x79, 0x75, 0x6f, 0x6b, 0x64, 0x5f, \
+                             0x59, 0x53, 0x4e, 0x49, 0x44, 0x3f, 0x3a, 0x35, 0x30, 0x2b, \
+                             0x25, 0x20, 0x1b, 0x17, 0x12, 0x0d, 0x09, 0x05, 0x02, 0x01, \
+                             0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+    /* ensure the given value is negative, since a CCA threshold > 0 is
+       just impossible: thus, any positive value given is considered
+       to be the absolute value of the actually wanted threshold */
+    if (value < 0) {
+        value = -value;
+    }
+
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_CCAEDTH, RSSI_value[value]);
+}
+
+void mrf24j40_set_option(mrf24j40_t *dev, uint16_t option, bool state)
+{
+    uint8_t tmp;
+
+    DEBUG("set option %i to %i\n", option, state);
+
+    /* set option field */
+    if (state) {
+        dev->netdev.flags |= option;
+        /* trigger option specific actions */
+        switch (option) {
+            case MRF24J40_OPT_CSMA:
+                DEBUG("[mrf24j40] opt: enabling CSMA mode (4 retries, macMinBE: 3 max BE: 5)\n");
+
+                /* Initialize CSMA seed with hardware address */
+                tmp = mrf24j40_reg_read_short(dev, MRF24J40_REG_TXMCR);
+                tmp &= 0b11100000;
+                tmp |= 0b00011000;
+                tmp |= 0b00000100;

```

Comment:
Are there not macros for this usage? It would be cool to see what's gonging on here.